mlir/lib/Transforms/lab4/sredneva_anastasiya/SrednevaMaxDepthPass.cpp

```diff
@@ -0,0 +1,57 @@
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+#include "mlir/Dialect/SCF/IR/SCF.h"
+#include "mlir/IR/BuiltinOps.h"
+#include "mlir/IR/Operation.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/IR/Region.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Tools/Plugins/PassPlugin.h"
+
+using namespace mlir;
+
+namespace {
+class SrednevaMaxDepthPass
+    : public PassWrapper<SrednevaMaxDepthPass, OperationPass<func::FuncOp>> {
+public:
+  StringRef getArgument() const final { return "SrednevaMaxDepthPass"; }
+  StringRef getDescription() const final {
+    return "Counts the max depth of region nests in the function.";
+  }
+  void runOnOperation() override {
+    getOperation()->walk([&](Operation *op) {
+      int maxDepth = 0;
+
+      std::function<void(Operation *, int)> getMaxDepth = [&](Operation *op,
+                                                              int depth) {
+        maxDepth = std::max(maxDepth, depth);
+        for (Region &region : op->getRegions()) {
+          for (Block &block : region) {
+            for (Operation &op2 : block) {
+              getMaxDepth(&op2, depth + 1);
+            }
+          }
+        }
+      };
+
+      getMaxDepth(op, 0);
```

Comment:
You are performing too many unnecessary operations by calling getMaxDepth on each operation. getMaxDepth already traverses all operations, no need to call it everywhere